finetune/network_modules.py

```diff
@@ -264,7 +264,37 @@ def regressor(hidden, targets, n_targets, config, train=False, reuse=None, **kwa
             'logits': outputs,
             'losses': loss
         }
+        
+def ordinal_regressor(hidden, targets, n_targets, config, train=False, reuse=None, **kwargs):
+    """
+    Ordinal Regressor using all-threshold loss.
 
+    :param hidden: The output of the featurizer. [batch_size, embed_dim]
+    :param targets: The placeholder representing the regression targets (binary threshold values). [batch_size]
+    :param n_targets: A python int containing the number of thresholds that the model should be learning to predict over.
+    :param dropout_placeholder:
+    :param config: A config object, containing all parameters for the featurizer.
+    :param train: If this flag is true, dropout and losses are added to the graph.
+    :param reuse: Should reuse be set within this scope.
+    :param kwargs: Spare arguments.
+    :return: dict containing:
+        logits: The regression outputs.
+        losses: All-threshold Loss for the regression targets.
+    """
+    with tf.variable_scope('ordinalregressor', reuse=reuse):
+        hidden = dropout(hidden, config.clf_p_drop, train)
+        outputs = perceptron(hidden, n_targets, config)
```

Comment:
We should split this into a matrix mult and an addition of a vector of biases so that we "tie weights"